packages/typography/src/elements/CodeBlock.tsx

```diff
@@ -28,13 +29,46 @@ interface IToken {
  */
 export const CodeBlock = React.forwardRef<HTMLPreElement, ICodeBlockProps>(
   (
-    { children, containerProps, highlightLines, isLight, isNumbered, language, size, ...other },
+    {
+      children,
+      containerProps,
+      highlightLines,
+      isLight,
+      isNumbered,
+      language = 'tsx',
+      size = 'medium',
+      ...other
+    },
     ref
   ) => {
     const containerRef = useRef<HTMLDivElement>(null);
     const code = (Array.isArray(children) ? children[0] : children) as string;
     const dependency = useMemo(() => [size, children], [size, children]);
     const containerTabIndex = useScrollRegion({ containerRef, dependency });
+    const [isPrismImported, setIsPrismImported] = useState(false);
+    const win = useWindow();
+
+    const importPrism = useCallback(async () => {
+      // TODO remove `importPrism` if/when `prismJS` releases v2 ESM
+      // https://github.com/orgs/PrismJS/discussions/3531
+      if (win && !isPrismImported) {
+        (win as any).Prism = Prism;
+
+        await Promise.all([
+          import('prismjs/components/prism-diff' as any),
+          import('prismjs/components/prism-json' as any)
+        ]);
```

Comment:
+1 to always have error states for dynamic imports

Reply:
We don't need an error state, per se. But we should always allow the component to render – whether or not it can be tokenized for language highlighting. The updated `try` .. `catch` .. `finally` allows for that.